File: Artificial Intelligence/family tree/grandchildren.pl
parent('Hasib', 'Rakib'). parent('Rakib', 'Sohel'). parent('Rakib', 'Rebeka'). parent('Rashid', 'Hasib'). grandparent(X,Z) :- parent(X, Y), parent(Y, Z).
findGc :- write('Grandparent : '), read(X), write('GrandChildren : '),
	grandparent(X, Gc), write(Gc), tab(5), fail.
findGc.

File: Artificial Intelligence/family tree/grandchildren.py
tupleList1=[('parent', 'Hasib', 'Rakib'),('parent', 'Rakib', 'Sohel'),
             		('parent', 'Rakib', 'Rebeka'),('parent', 'Rashid', 'Hasib')]

              # Procedure to find the grandchildren of X

X=str(input("Grandparent:"))
print('Grandchildren:', end=' ')
i=0
while(i<=3):
    if ((tupleList1[i][0] == 'parent')&( tupleList1[i][1] == X)):
        for j in range(4):
            if ((tupleList1[j][0] == 'parent') & ( tupleList1[i][2] == tupleList1[j][1])):
                print(tupleList1[j][2], end=' ')   
    i=i+1  
